### ISSUE TITLE
Feature: When comparison types are compatible, but not ideal, report a different finding that is LOW severity

### ISSUE THREAD

**Issue by danielpops** [duo-labs/parliament#29]
A policy like => 

```
{
    "Statement": [
        {
            "Action": [
                "ec2:TerminateInstances"
            ],
            "Effect": "Allow",
            "Resource": "*",
            "Condition": {
                "StringEquals": {
                    "ec2:InstanceProfile": "arn:aws:iam::123456789012:instance-profile/my_role"
                }
            }
        }
    ]
}
```

Results in a finding =>
```
{"issue": "MISMATCHED_TYPE", "title": "Mismatched type", "severity": "MEDIUM", "detail": "Type mismatch: StringEquals requires a value of type String but given Arn", "location": {"location": {"ec2:InstanceProfile": "arn:aws:iam::123456789012:instance-profile/my_role"}}}
```

I *think* this is an incorrect finding, because I was expecting to be able to use StringEquals for a field that contains an ARN, but I could very well be mistaken. Raising the issue to see your thoughts.

**Comment by danielpops** [comment]
I think maybe this is an AWS documentation issue for the ec2:InstanceProfile condition key. AFAICT, the only condition keys available for ARN operators are `aws:PrincipalArn` and `aws:SourceArn`, neither of these are applicable for this scenario:

`aws:PrincipalArn` => `Use this key to compare the Amazon Resource Name (ARN) of the principal that made the request with the ARN that you specify in the policy`
`aws:SourceArn` => `Use this key to compare the source of the request with the Amazon Resource Name (ARN) that you specify in the policy`

**Comment by 0xdabbad00** [comment]
So it is correct that you can use a string comparison an an ARN, but it is arguably "more correct" to use an ARN comparison operator on it.  This is a finding type I should break up into `Info` level concerns for the example you provided, and then maybe keep it at `Medium` for situations where someone is using a Numeric operator against a string, which would actually be incorrect (ie. it would not work).

**Comment by danielpops** [comment]
Can you suggest what the policy should ideally look like in this case? E.g. how to do an arn comparison for this field?

**Comment by 0xdabbad00** [comment]
I think in this case, it's just a matter of doing `ArnEquals` instead of `StringEquals`.

**Comment by 0xdabbad00** [comment]
I tested and using `ArnEquals` instead of `StringEquals` does solve the problem of avoiding having a Finding reported.

I've changed the title of this ticket to make this an enhancement to not report a Medium severity finding under these circumstances where the String comparison is compatible with the operators, even though an Arn comparison would be more ideal.  In this situation, a Low severity finding should be reported that people can choose to mute as needed.
